### DISCUSSION TITLE
Gauge vs. Robot Framework

### DISCUSSION THREAD

**Discussion by wlad** [getgauge/gauge#1953] [General]
I miss one important thing in Gauge's docs:

## Why and when should I use Gauge instead of Robot Framework?

Does anybody here has experience with both frameworks [Gauge](https://gauge.org/) and [Robot Framework](https://robotframework.org/) (another open source (test) automation framework) and can share his thoughts about PROS and CONS? May be anybody used Robot in the past an switched to Gauge or vise versa? Or may be somebody is recently comparing both frameworks? Please share your thoughts.

`#robotframework`
`#robotframework vs gauge`

**Comment by zabil** [comment]
Based on my conversations with a few people who've migrated from Robot to Gauge (and my own biased observations). 

Disclaimer: Last time I looked and talked about Robot was 3 years ago. Things should have changed.

* Robot framework works only on Python while Gauge supports other langauges
* People prefer Gauge reporting as based in the specification format as it was closed to business langauge vs Robot frameworks keyword driven approach
* IDE support for editing the specification seemed better at that time as Gauge specifications are in Markdown vs custom structure of Robot
* People preferred the VS Code IDE plugin integration of Gauge and Python
* Simpler and native Data driven testing support using markdown tables and external csv files
* Markdown syntax was easier to maintain than the keyword driven syntax of Robot
* Gauge has re-usability features for steps like [aliases](https://docs.gauge.org/writing-specifications.html?os=linux&language=javascript&ide=vscode#step-alias) and [concepts](https://docs.gauge.org/writing-specifications.html?os=linux&language=javascript&ide=vscode#concepts)
* An easier way to write and manage plugins to extend Gauge's funtionality. Also the ability to write plugins in any language.

I hope this makes sense.
